### ISSUE TITLE
Телеграмм Бот Исследовательская часть

### ISSUE THREAD

**Issue by DaniilPopov52** [DaniilPopov52/-#9]

**Comment by DaniilPopov52** [comment]
Добрый день, я попробовал сделать предварительную запись в МФЦ а её нельзя сделать без входа в аккаунт, это значит что мне в своём боте нужно запрашивать с пользователя Логин и Пароль?

**Comment by DaniilPopov52** [comment]
Как вошёл в аккаунт нужно выбрать филиал для записи, затем нужно выбрать по какому вопросу хотите записаться на приём, затем выбрать свободную дату и свободное время, а затем вам нужно дописать недостающую информацию о себе (например серию и номер паспорта, номер дела, адрес проживание и тп.) и на этом запись заканчивается

**Comment by wldmr-key** [comment]
Если телеграм поделился номером телефона (или еще каким-либо идентификатором), то тв можешь установить клиента и аутентифицироваться ему не нужно. Подумай, как сделать этот процесс безопасным. Поставь себя на место клиента - как тебе было бы удобнее записываться

**Comment by DaniilPopov52** [comment]
Конечно удобней если не надо аутентифицироваться

**Comment by wldmr-key** [comment]
Замечание `Поставь себя на место клиента - как тебе было бы удобнее записываться` касалось всех действий клиента

**Comment by DaniilPopov52** [comment]
Удобней было бы просто зайти в бота, выбрать своё мфц в который хочу записаться на приём, выбрать дату и время и вписать требуемую информацию

**Comment by wldmr-key** [comment]
Ты точно исследовал все действия клиента? Что значит "требуемую информацию"?

**Comment by DaniilPopov52** [comment]
Требуемая информация это серия номер паспорта, номер дела, адрес проживания и так далее

**Comment by wldmr-key** [comment]
то есть, тебе, как клиенту реально хочется это вводить?!!!

**Comment by DaniilPopov52** [comment]
Конечно нет, но как это обойти если на сайте МФЦ требуют это вводить

**Comment by wldmr-key** [comment]
ну, подумай!!!! Если у тебя там есть какие-то дела, то ты уже ассоциирован со своим номером телефона. Зачем тебе брать пример с их дурацкого сайта?! Сделай как для себя!

**Comment by DaniilPopov52** [comment]
Для себя удобней просто зайти в бота и выбрать мфц в который хочу прийти ,и дату и время прибытия

**Comment by wldmr-key** [comment]
Еще раз - Ты точно исследовал все действия клиента?

**Comment by DaniilPopov52** [comment]
Ну да, клиент выбирает мфц, выбирает "с каким вопросом хочет обратиться в мфц", выбирает день и время и вписывает информацию о себе

**Comment by DaniilPopov52** [comment]
По крайней мере такая запись на сайте мфц

**Comment by DaniilPopov52** [comment]
Добрый день, а можно написать не бота а мобильное приложение ? Если не мфц то какой нибудь доставки или чего нибудь подобного ? Я просто подумал и понял что это для меня более интересно

**Comment by wldmr-key** [comment]
Меня тревожит `какой-нибудь доставки`. Предыдущий опыт говорит мне, что у тебя большие проблемы с изучением предметной области. Попробуй, конечно, если хочешь

**Comment by DaniilPopov52** [comment]
Например, мобильное приложение магазина одежды. Это пойдёт как курсовая а потом дипломная работа ?

**Comment by wldmr-key** [comment]
а, зачем магазину одежды мобильное приложение? ))
Объяснишь - пойдет

**Comment by DaniilPopov52** [comment]
Это как интернет магазин, чтобы упростить покупку одежды

**Comment by DaniilPopov52** [comment]
Чтобы можно было посмотреть и заказать товар не выходя из дома

**Comment by wldmr-key** [comment]
пока не убедительно, но можешь попробовать

**Comment by DaniilPopov52** [comment]
А мобильное приложение подойдёт для дипломной работы ?

**Comment by wldmr-key** [comment]
а, чем оно хуже немобильного ))

**Comment by DaniilPopov52** [comment]
Ну в принципе ничем)

**Comment by DaniilPopov52** [comment]
Добрый вечер, можете пожалуйста посмотреть ТЗ к моей новой теме

**Comment by DaniilPopov52** [comment]
[Тех.Задание.docx](https://github.com/DaniilPopov52/-/files/5323972/default.docx)

**Comment by wldmr-key** [comment]
Очередной раз... Назначение разработки - для кого она предназначена? Кто за нее заплатит? Несчастные покупатели одежды? Или, все-таки, магазин, который продает одежду? Тогда у него другая функция...

переделывай

**Comment by DaniilPopov52** [comment]
Ну получается что назначение - удобство пользователю купить одежду не выходя из дома.
И цель это увеличение дохода магазина и своевременное информирование о изменении товаров.

**Comment by DaniilPopov52** [comment]
Ну предназначено приложения для пользователей желающих купить одежду, а заплатит мне за это приложение магазин одежды. Но я все равно не понимаю какая функция у него, по идее же его функция это оформление заказа одежды, чтобы у магазина вырос доход

**Comment by wldmr-key** [comment]
я тебе русским языком написал - кто заплатит (пусть, в воображении) за разработку? За что заплатит? Из каких денег?
Вот, за что заплатит, то и будет назначением
А, как будет осуществляться, то, за что заплатит, то и будет функциями
А, слово "удобство" здесь вообще не при чем...

**Comment by DaniilPopov52** [comment]
Получается что, назначение - увеличение дохода магазина.
Функциональность - оформление заказов одежды не выходя из дома.

**Comment by wldmr-key** [comment]
только более конкретно
посредством чего должны увеличиться доходы магазина? Они там еще блэкджек устроят?

Функциональность - какие конкретно функции позволит осуществлять приложение (для клиента и для магазина)

**Comment by DaniilPopov52** [comment]
Увеличится доход магазина посредством увеличения клиентов, так как удобней заказать в мобильном приложении сидя дома, чем идти в магазин.


Функции для магазина:
1) Добавление и изменение товара

Функции для клиента:
1) Оформление заказа
2) Поиск товара по критериям

**Comment by wldmr-key** [comment]
неважно пока
Начинай исследовательскую часть, может, тогда, для тебя эти вещи станут яснее...

**Comment by DaniilPopov52** [comment]
Добрый день, посмотрите пожалуйста мою исследовательскую часть

**Comment by DaniilPopov52** [comment]
[Исследовательская часть.docx](https://github.com/DaniilPopov52/-/files/5341768/default.docx)

**Comment by wldmr-key** [comment]
. Постановка задачи - нет, не так
Описание предметной области - нет. Продажа одежды существует и без твоего приложения
Анализ аналогов - никакой, так как нет постановки собственной задачи и, следовательно, сравнивать не с чем.

**Comment by DaniilPopov52** [comment]
А какая тогда постановка задачи ? "Разработать мобильное приложение для продажи одежды"?

**Comment by wldmr-key** [comment]
очевидно, в исследовательской части нет смысла дублировать постановку задачи из тз. Какие задачи у тебя в исследовании?

**Comment by DaniilPopov52** [comment]
Исследовать предметную область и аналоги мобильного приложения магазина одежды

**Comment by DaniilPopov52** [comment]
[Исследовательская часть.docx](https://github.com/DaniilPopov52/-/files/5377116/default.docx)

**Comment by wldmr-key** [comment]
Описание предметной области - нет, все не так. Продажа одежды существует и без твоего приложения. Одежду покупают не пользователи, а покупатели. У одежды нет id. 

Пожалуйста, раздели предметную область (продажа одежды) и свое будущее приложение!!! Иначе оно никогда не появится! 

Аналоги - никакие (точнее анализ никакой), концептуальная схема упала с неба, сценарии использования уже даже Мища исправил - спроси у него почему ))

У тебя все будет на Android Studio? А, сервер?

**Comment by DaniilPopov52** [comment]
Можно я исследовательскую часть тоже разделю на части чтобы было попроще

**Comment by DaniilPopov52** [comment]
[Описание предметной области.docx](https://github.com/DaniilPopov52/-/files/5380605/default.docx)

**Comment by DaniilPopov52** [comment]
Посмотрел пример в интернете и вот что получилось

**Comment by wldmr-key** [comment]
> Можно я исследовательскую часть тоже разделю на части чтобы было попроще

конечно. И лучше вынести в отдельные issues,  а то здесь у нас много обсуждения не по делу

> Посмотрел пример в интернете и вот что получилось

Можешь брать его в кавычки и вставлять в подраздел описание работы _обычных_ интернет магазинов. Но твоя предметная область - магазин одежды...

**Comment by DaniilPopov52** [comment]
Я вынес в отдельное issue
